src/Datadog.Trace/Datadog.Trace.csproj

```diff
@@ -46,11 +46,15 @@
     <DefineConstants>$(DefineConstants);ASYNCLOCAL;HASHTABLE</DefineConstants>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'netcoreapp3.1' ">
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
     <DefineConstants>$(DefineConstants);ASYNCLOCAL;HASHTABLE</DefineConstants>
   </PropertyGroup>
 
-  <!-- Remove System.Collections.NonGeneric. This was defined in Serilog 2.8.0 but it is unnecessary -->
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1' ">
+    <DefineConstants>$(DefineConstants);ASYNCLOCAL;HASHTABLE;FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN</DefineConstants>
```

Comment:
I had to split the definition of netstandard2.0 and netcoreapp3.1.
The original has netstandard2.1, which makes me wonder if we should start to compile against that and start taking advantage of spans throughout the main code base.

Reply:
As we discussed, netstandard2.1 doesn't give us anything over .NET Core 3.0, as we don't currently intend to support Mono or Xamarin, so I think this is the best option for now